components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,538 @@
+
+	return match
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processMap(map)
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+	map = mapFunctions.getTournamentVars(map)
+	map = mapFunctions.getParticipantsData(map)
+
+	return map
+end
+
+function CustomMatchGroupInput.processOpponent(record, date)
+	local opponent = Opponent.readOpponentArgs(record)
+		or Opponent.blank()
+
+	-- Convert byes to literals
+	if opponent.type == Opponent.team and opponent.template:lower() == 'bye' then
+		opponent = {type = Opponent.literal, name = 'BYE'}
+	end
+
+	Opponent.resolve(opponent, date)
+	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+
+	--score2 & score3 support for every match
+	local score2 = tonumber(record.score2 or '')
+	local score3 = tonumber(record.score3 or '')
+	if score2 then
+		record.extradata = {
+			score2 = score2,
+			score3 = score3,
+			set1win = Logic.readBool(record.set1win),
+			set2win = Logic.readBool(record.set2win),
+			set3win = Logic.readBool(record.set3win),
+			additionalScores = true
+		}
+	end
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processPlayer(player)
+	return player
+end
+
+--
+--
+-- function to sort out winner/placements
+function CustomMatchGroupInput._placementSortFunction(table, key1, key2)
+	local op1 = table[key1]
+	local op2 = table[key2]
+	local op1norm = op1.status == _STATUS_HAS_SCORE
+	local op2norm = op2.status == _STATUS_HAS_SCORE
+	if op1norm then
+		if op2norm then
+			local op1setwins = CustomMatchGroupInput._getSetWins(op1)
+			local op2setwins = CustomMatchGroupInput._getSetWins(op2)
+			if op1setwins + op2setwins > 0 then
+				return op1setwins > op2setwins
+			else
+				return tonumber(op1.score) > tonumber(op2.score)
+			end
+		else return true end
+	else
+		if op2norm then return false
+		elseif op1.status == _STATUS_DEFAULT_WIN then return true
+		elseif Table.includes(ALLOWED_STATUSES, op1.status) then return false
+		elseif op2.status == _STATUS_DEFAULT_WIN then return false
+		elseif Table.includes(ALLOWED_STATUSES, op2.status) then return true
+		else return true end
+	end
+end
+
+function CustomMatchGroupInput._getSetWins(opp)
+	local extradata = opp.extradata or {}
+	local set1win = extradata.set1win and 1 or 0
+	local set2win = extradata.set2win and 1 or 0
+	local set3win = extradata.set3win and 1 or 0
+	local sum = set1win + set2win + set3win
+	return sum
+end
+
+--
+-- match related functions
+--
+function matchFunctions.readDate(matchArgs)
+	return matchArgs.date
+		and MatchGroupInput.readDate(matchArgs.date)
+		or {
+			date = MatchGroupInput.getInexactDate(globalVars:get('tournament_enddate')),
+			dateexact = false,
+		}
+end
+
+function matchFunctions.getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', '2v2'))
+	match.showh2h = Logic.emptyOr(match.showh2h, Variables.varDefault('showh2h'))
```

Comment:
is h2h stuff a thing on this wiki?

Reply:
To be implemented, but I think we should leave it because it will be useful